File: front/src/dto/AttendanceDto.ts
import type {AttendeeDto} from "@/dto/AttendeeDto.ts";
import type {TimeStampDto} from "@/dto/TimeStampDto.ts";

export class AttendanceDto {

  public date : TimeStampDto
  public attendee : AttendeeDto

  public constructor(date : TimeStampDto, attendee : AttendeeDto) {
    this.date = date;
    this.attendee = attendee;
  }


}

File: back/src/main/java/be/naaturel/letsmeet/dto/httpRequest/AttendanceDTO.java
package be.naaturel.letsmeet.dto.httpRequest;

import java.util.Objects;

public class AttendanceDTO {
    public EventDateDTO date;
    public AttendeeDTO attendee;

    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;
        AttendanceDTO a = (AttendanceDTO) obj;
        return a.date.timestamp == ((AttendanceDTO) obj).date.timestamp &&
                a.attendee.name == ((AttendanceDTO) obj).attendee.name;
    }

    @Override
    public int hashCode() {
        return Objects.hash(date, attendee);
    }
}
